ark/util/serialize.ts

```diff
@@ -95,7 +94,7 @@ const _serialize = (
 		case "symbol":
 			return printableOpts.onSymbol(data as symbol)
 		case "bigint":
-			return `$ark.bigint-${data}`
+			return opts.onBigInt?.(data as bigint) ?? `$ark.bigint-${data}`
```

Comment:
The key is here and in the undefined case we need to return what we did before this PR (just `5n` or similar in the case of bigint) by default, since this function is also used for error messages.

In the root of the `snapshot` function specifically, when we pass `opts` to `_serialize`, we should instead pass

```ts
{onUndefined: () => `$ark.undefined`, onBigInt: n => `${bigintPrefix}${n}`, ...opts }
```

Or similar.

Reply:
ah! updated